### PR TITLE
Add aggregations and filtering to OC plugins

### PR DESCRIPTION
**Pull request by nik-kc**
## What does this PR change?
* 

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: pkg/customcost/matcher.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	switch CustomCostProperty(identifier.Field.Name) {
+	case CustomCostZoneProp:
+		return cc.Zone, nil
+	case CustomCostAccountNameProp:
+		return cc.AccountName, nil
+	case CustomCostChargeCategoryProp:
+		return cc.ChargeCategory, nil
+	case CustomCostResourceTypeProp:
+		return cc.ResourceType, nil
+	case CustomCostDomainProp:
```
Comment (ameijer) [inline pkg/customcost/matcher.go:49]: can you also add Description, resourceName, providerID, and usageUnit?

File: pkg/customcost/types.go [hunk 2]
```diff
@@ -197,15 +197,28 @@ func (ccs *CustomCostSet) Aggregate(aggregateBy []string) error {
...
+		var aggKey string
+		if agg == CustomCostZoneProp {
+			aggKey = cc.Zone
+		} else if agg == CustomCostAccountNameProp {
+			aggKey = cc.AccountName
+		} else if agg == CustomCostChargeCategoryProp {
+			aggKey = cc.ChargeCategory
+		} else if agg == CustomCostResourceTypeProp {
+			aggKey = cc.ResourceType
+		} else if agg == CustomCostDomainProp {
```
Comment (ameijer) [inline pkg/customcost/types.go:220]: can you also add support for description, resourceName, providerID, and usageUnit?

**Review by ameijer — COMMENTED** [review]
looks good, wanted to see if you would be willing to add support for the remainder of the attributes?

**Review by ameijer — APPROVED** [review]
LGTM

**Review by teevans — APPROVED** [review]

**Pull request merged**